egs2/TEMPLATE/asr1/pyscripts/utils/evaluate_pseudomos.py

```diff
@@ -0,0 +1,196 @@
+                "constant",
+                constant_values=0,
+            )
+        gen_xs = torch.from_numpy(np.stack(gen_xs)).to(device)
+        # Compute pseudo MOS.
+        scores = predictor(gen_xs, fs)
+        for gen_path, score in zip(gen_paths, scores):
+            gen_basename = _get_basename(gen_path)
+            pmos = score.item()
+            logging.info(f"{gen_basename} {pmos:.4f}")
+            pmos_dict[gen_basename] = pmos
+    return pmos_dict
+
+
+def get_parser() -> argparse.Namespace:
+    """Get argument parser."""
+    parser = argparse.ArgumentParser(description="Evaluate pseudo MOS.")
+    parser.add_argument(
+        "gen_wavdir_or_wavscp",
+        type=str,
+        help="Path of directory or wav.scp for generated waveforms.",
+    )
+    parser.add_argument(
+        "--outdir",
+        type=str,
+        help="Path of directory to write the results.",
+    )
+    parser.add_argument("--device", type=str, default="cuda:0", help="Inference device")
+    parser.add_argument(
+        "--batchsize",
+        default=4,
+        type=int,
+        help="Number of batches.",
+    )
+    parser.add_argument(
+        "--verbose",
+        default=1,
+        type=int,
+        help="Verbosity level. Higher is more logging.",
+    )
+    return parser
+
+
+def main():
+    """Run pseudo MOS calculation in parallel."""
+    args = get_parser().parse_args()
+
+    # logging info
+    if args.verbose > 1:
+        logging.basicConfig(
+            level=logging.DEBUG,
+            format="%(asctime)s (%(module)s:%(lineno)d) %(levelname)s: %(message)s",
+        )
+    elif args.verbose > 0:
+        logging.basicConfig(
+            level=logging.INFO,
+            format="%(asctime)s (%(module)s:%(lineno)d) %(levelname)s: %(message)s",
+        )
+    else:
+        logging.basicConfig(
+            level=logging.WARN,
+            format="%(asctime)s (%(module)s:%(lineno)d) %(levelname)s: %(message)s",
+        )
+        logging.warning("Skip DEBUG/INFO messages")
+
+    # Find files for generated speech waveforms.
+    if os.path.isdir(args.gen_wavdir_or_wavscp):
+        gen_files = sorted(find_files(args.gen_wavdir_or_wavscp))
+    else:
+        with open(args.gen_wavdir_or_wavscp) as f:
+            gen_files = [line.strip().split(None, 1)[1] for line in f.readlines()]
+        if gen_files[0].endswith("|"):
+            raise ValueError("Not supported wav.scp format.")
+
+    # Get and divide list
+    if len(gen_files) == 0:
+        raise FileNotFoundError("Not found any generated audio files.")
+    logging.info("The number of utterances = %d" % len(gen_files))
+
+    if torch.cuda.is_available() and ("cuda" in args.device):
+        device = torch.device(args.device)
+    else:
+        device = torch.device("cpu")
+
+    # Load predictor for UTMOS22.
+    predictor = torch.hub.load(
+        "tarepan/SpeechMOS:v1.1.0", "utmos22_strong", trust_repo=True
+    ).to(device)
```

Comment:
It is a very cool add-on. Do you also consider support other audo-MOS evaluation tools in the future?

Reply:
Currently I am not going to add it, but adding other MOS prediction tools sounds good.
So I have added a flag `mos_pred_tool` to switch the MOS prediction toolkits.